src/runtime_src/core/edge/user/aie/common_layer/adf_runtime_api.cpp

```diff
@@ -680,4 +710,224 @@ err_code gmio_api::wait()
     return err_code::ok;
 }
 
+/************************************ dma_api ************************************/
+
+static uint8_t relativeToAbsoluteRow(int tileType, uint8_t row)
+{
+    uint8_t absoluteRow = row;
+    if (tileType == 0) //aie tile
+        absoluteRow += (1 + config_manager::s_num_reserved_rows);
+    else if (tileType == 2) //memory_tile
+        absoluteRow += 1;
+    return absoluteRow;
+}
+
+err_code dma_api::configureBdWaitQueueEnqueueTask(int tileType, uint8_t column, uint8_t row, int dir, uint8_t channel, uint32_t repeatCount, bool enableTaskCompleteToken, std::vector<uint8_t> bdIds, std::vector<dma_api::buffer_descriptor> bdParams)
+{
+    int status = (int)err_code::ok;
+
+    if (config_manager::s_pDevInst->DevProp.DevGen == XAIE_DEV_GEN_AIE)
+        return errorMsg(err_code::internal_error, "ERROR: adf::dma_api::enqueueTask: Does not support AIE architecture.");
+
+    if (bdParams.empty())
+        return errorMsg(err_code::internal_error, "ERROR: adf::dma_api::enqueueTask: Empty buffer descriptors.");
+
+    if (bdIds.size() != bdParams.size())
+        return errorMsg(err_code::internal_error, "ERROR: adf::dma_api::enqueueTask: The number of BD IDs and the number of BDs are different.");
+
+    // configure BD
+    for (int i = 0; i < bdParams.size(); i++)
+        status |= (int)configureBD(tileType, column, row, bdIds[i], bdParams[i]);
+
+    //wait task queue
+    status |= (int)waitDMAChannelTaskQueue(tileType, column, row, dir, channel);
+
+    //start queue
+    status |= (int)enqueueTask(tileType, column, row, dir, channel, repeatCount, enableTaskCompleteToken, bdIds[0]);
+
+    return (err_code)status;
+}
+
+err_code dma_api::configureBD(int tileType, uint8_t column, uint8_t row, uint8_t bdId, const dma_api::buffer_descriptor& bdParam)
+{
+    int driverStatus = XAIE_OK; //0
+    debugMsg(static_cast<std::stringstream &&>(std::stringstream() << "dma_api::configureBD" << std::endl).str());
```

Comment:
Ouch!
```suggestion
    debugMsg(std::move(std::stringstream() << "dma_api::configureBD" << std::endl).str());
```
but I do not understand why you need this obfuscation.
And what about defining a version of `debugMsg` taking a `const std::stringstream` at the first place so you can write
```c++
    debugMsg(std::stringstream {} << "dma_api::configureBD" << std::endl);
```
or here just
```c++
    debugMsg("dma_api::configureBD\n");
```
or even
```c++
    debugMsgNL("dma_api::configureBD");
```
At the end if it is not simple, there is an issue in the API...